src/mmw/js/src/modeling/models.js

```diff
@@ -221,6 +221,54 @@ var ScenariosCollection = Backbone.Collection.extend({
             }
             model.set('active', active);
         });
+    },
+
+    duplicateScenario: function(cid) {
+        var source = this.get(cid),
+            sourceMods = source.get('modifications').models,
+            newModel = new ScenarioModel({
+                is_current_conditions: false,
+                name: this.makeNewScenarioName('Copy of ' + source.get('name'))
+            });
+
+        newModel.set('modifications', new ModificationsCollection(sourceMods, {
+            model: ModificationModel
+        }));
```

Comment:
Consider merging this logic with the initialization of `newModel` above. Also, we shouldn't need the second argument.